base/collection_n1ql.go

```diff
@@ -119,8 +119,57 @@ func (c *Collection) CreatePrimaryIndex(indexName string, options *N1qlIndexOpti
 	return CreatePrimaryIndex(c, indexName, options)
 }
 
-func (c *Collection) WaitForIndexOnline(indexName string) error {
-	return WaitForIndexOnline(c, indexName)
+// WaitForIndexesOnline takes set of indexes and watches them till they're online.
+func (c *Collection) WaitForIndexesOnline(indexNames []string, failfast bool) error {
+	logCtx := context.TODO()
+	mgr := c.cluster.QueryIndexes()
+	maxNumAttempts := 180
+	if failfast {
+		maxNumAttempts = 1
+	}
+	retrySleeper := CreateMaxDoublingSleeperFunc(maxNumAttempts, 100, 5000)
+	retryCount := 0
+
+	onlineIndexes := make(map[string]bool)
+
+	indexOption := gocb.GetAllQueryIndexesOptions{
+		ScopeName:      c.ScopeName(),
+		CollectionName: c.Name(),
```

Comment:
one last change I think to add noretries to this strategy becuase we'll handle retries on our own retrysleeper

Reply:
I have added the fail fast parameter here already to allow for it to only execute once.

Reply:
The failfast parameter is different than the timeout.

All kv ops use timeouts and retry strategy based on what they inherit when you create the cluster. As per https://github.com/couchbase/sync_gateway/commit/46b733faf1180db7c055bd97a421f5c4e1e9ea0d

we retry operations up to `DefaultGocbV2OperationTimeout = 10 * time.Second` or optionally overriden by `BucketOpTimeoutMs` in db config.

Reply:
Ah i get you, think I again misunderstood what you meant sorry